src/Components/Components/src/Binding/IHostEnvironmentFormDataProvider.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Binding;
+
+/// <summary>
+/// An interface implemented by <see cref="FormDataProvider"/> classes that can receive
+/// the form data from the host environment.
+/// </summary>
+public interface IHostEnvironmentFormDataProvider
```

Comment:
I understand pulling form data from the HttpContext, but what's the scenario for reading it from the environment?

Reply:
This is the same pattern we follow for navigation manager and other types in `Microsoft.AspNetCore.Components`. Unfortunately, we need to abstract the concept from Http, and the name, and xml docs follow the same convention as those other types.